python/ray/serve/_private/api.py

```diff
@@ -185,46 +179,66 @@ def serve_start(
     else:
         controller_name = format_actor_name(get_random_letters(), SERVE_CONTROLLER_NAME)
 
-    if isinstance(http_options, dict):
-        http_options = HTTPOptions.parse_obj(http_options)
-    if http_options is None:
-        http_options = HTTPOptions()
-
     # Used for scheduling things to the head node explicitly.
     # Assumes that `serve.start` runs on the head node.
     head_node_id = ray.get_runtime_context().node_id.hex()
-    controller = ServeController.options(
-        num_cpus=1 if dedicated_cpu else 0,
-        name=controller_name,
-        lifetime="detached" if detached else None,
-        max_restarts=-1,
-        max_task_retries=-1,
+    controller_start_options = {
+        "num_cpus": 1 if dedicated_cpu else 0,
+        "name": controller_name,
+        "lifetime": "detached" if detached else None,
+        "max_restarts": -1,
+        "max_task_retries": -1,
         # Schedule the controller on the head node with a soft constraint. This
         # prefers it to run on the head node in most cases, but allows it to be
         # restarted on other nodes in an HA cluster.
-        scheduling_strategy=NodeAffinitySchedulingStrategy(head_node_id, soft=True)
+        "scheduling_strategy": NodeAffinitySchedulingStrategy(head_node_id, soft=True)
         if RAY_INTERNAL_SERVE_CONTROLLER_PIN_ON_NODE
         else None,
-        namespace=SERVE_NAMESPACE,
-        max_concurrency=CONTROLLER_MAX_CONCURRENCY,
-    ).remote(
-        controller_name,
-        http_config=http_options,
-        head_node_id=head_node_id,
-        detached=detached,
-    )
+        "namespace": SERVE_NAMESPACE,
+        "max_concurrency": CONTROLLER_MAX_CONCURRENCY,
+    }
+
+    if os.environ.get("EXPERIMENT_DISABLE_HTTP_PROXY", "0") == "1":
```

Comment:
move this to a global constant flag. you can find similar pattern in other files of this code base.

Reply:
and prefix it with SERVE, and change the `EXPERIMENT` to `EXPERIMENTAL`